speakrune/gen/speakrune.hoon

```diff
@@ -0,0 +1,51 @@
+|=  raw=tape
+^-  tape
+=<  ?:((valid raw) (turn raw convert) "input contains non-rune characters")
+|%
```

Comment:
For constructions like this, you could use `|^`. Here, that would look like:

```hoon
|^  ?:  (valid raw)
      (turn raw convert)
    "input contains non-rune characters"
::
++  valid
  etc
--
```

Reply:
That's handy. Would it be good practice to still cast to a tape at the start?
```
|^  ^-  tape
    ?: ...
```

Reply:
Yeah, casting's always good. You can do that above the `|^` or within it, doesn't really matter. Though having it directly under the `|=` is the most common style.